google/cloud/bigquery/big_query_read_connection.gcpcxx.pb.h

```diff
@@ -0,0 +1,102 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Generated by the Codegen C++ plugin.
+// If you make any local changes, they will be lost.
+// source: google/cloud/bigquery/storage/v1/storage.proto
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_BIGQUERY_BIG_QUERY_READ_CONNECTION_GCPCXX_PB_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_BIGQUERY_BIG_QUERY_READ_CONNECTION_GCPCXX_PB_H
+
+#include "google/cloud/bigquery/big_query_read_connection_idempotency_policy.gcpcxx.pb.h"
+#include "google/cloud/bigquery/internal/big_query_read_stub.gcpcxx.pb.h"
+#include "google/cloud/bigquery/retry_traits.h"
+#include "google/cloud/backoff_policy.h"
+#include "google/cloud/connection_options.h"
+#include "google/cloud/status_or.h"
+#include "google/cloud/stream_range.h"
+#include "google/cloud/version.h"
+#include <memory>
+
+namespace google {
+namespace cloud {
+namespace bigquery {
+inline namespace GOOGLE_CLOUD_CPP_GENERATED_NS {
+
+struct BigQueryReadConnectionOptionsTraits {
+  static std::string default_endpoint();
+  static std::string user_agent_prefix();
+  static int default_num_channels();
+};
+
+using BigQueryReadConnectionOptions =
+    google::cloud::ConnectionOptions<BigQueryReadConnectionOptionsTraits>;
+
+using BigQueryReadRetryPolicy = google::cloud::internal::TraitBasedRetryPolicy<
+    bigquery_internal::BigQueryReadRetryTraits>;
+
+using BigQueryReadLimitedTimeRetryPolicy =
+    google::cloud::internal::LimitedTimeRetryPolicy<
+        bigquery_internal::BigQueryReadRetryTraits>;
+
+using BigQueryReadLimitedErrorCountRetryPolicy =
+    google::cloud::internal::LimitedErrorCountRetryPolicy<
+        bigquery_internal::BigQueryReadRetryTraits>;
+
+void BigQueryReadReadRowsStreamingUpdater(
+    ::google::cloud::bigquery::storage::v1::ReadRowsResponse const& response,
+    ::google::cloud::bigquery::storage::v1::ReadRowsRequest& request);
+
+class BigQueryReadConnection {
+ public:
+  virtual ~BigQueryReadConnection() = 0;
+
+  virtual StatusOr<::google::cloud::bigquery::storage::v1::ReadSession>
+  CreateReadSession(
+      ::google::cloud::bigquery::storage::v1::CreateReadSessionRequest const&
+          request);
+
+  virtual StreamRange<::google::cloud::bigquery::storage::v1::ReadRowsResponse>
+  ReadRows(::google::cloud::bigquery::storage::v1::ReadRowsRequest request);
```

Comment:
As we discussed offline, I think this argument should be passed by `const&`, just like all the other arguments here. Especially since this is an interface that our customers will implement, I don't think we should require all possible subclasses to accept a copy. And this is a small proto, so I doubt there's any performance gain to be had by moving.